keytree/src/lib.rs

```diff
@@ -37,6 +41,41 @@ impl Xprv {
     }
 }
 
+/// Returns a intermediate child pubkey.
+pub fn intermediate_key(
+    xpub: &Xpub,
+    label: &'static std::string::String,
```

Comment:
Instead of passing a label, we should pass a closure `F: FnOnce(&mut Transcript)` where the user can commit arbitrary key-value pairs as a "derivation path". Note: the user may commit more than one labeled input.

Reply:
The closure is supposed to be called by keytree in-between committing the xpub (`pt`,`dk`) and squeezing the blinding factor `f`.